### ISSUE TITLE
add render timeout 

### ISSUE THREAD

**Issue by maxisam** [angular/angular#51549]
# 🚀 Feature request

#### What modules are relevant for this feature request?

- [ ] aspnetcore-engine
- [ ] builders
- [X] common
- [X] express-engine
- [ ] hapi-engine

### Description
Should have a way to set timeout. So if angular universal render more than certain time, it should throw a timeout error.
It is highly possible certain links in the backend are broken at some point, without timeout it will be hanging there.

I think  angular/universal#863 can be handled by this feature

 A clear and concise description of the problem or missing capability...

### Describe the solution you'd like

Build-in a timeout function like https://github.com/expressjs/timeout/blob/master/index.js

### Describe alternatives you've considered

 Have you considered any alternative solutions or workarounds?

**Comment by alan-agius4** [comment]
@AndrewKushnir do you think this should be part of `@angular/platform-server` or `@angular/ssr`?

**Comment by AndrewKushnir** [comment]
@alan-agius4 I think we should consider adding this config option to the `renderApplication` and `renderModule` functions (within `@angular/platform-server`) and potentially also expose them via APIs from `@angular/ssr`.

**Comment by Platonn** [comment]
Hi @alan-agius4 @AndrewKushnir 
IMHO we should take care not only of rendering timeouts but also any custom error propagation from Angular app to Express in SSR (or worker in case of SSG).

Currently, when any async error happens in a custom Angular app (e.g. unhandled TypeError 'cannot use property x of undefined' or we receive a HTTP error response from the upstream backend API), we have no way OOTB in Angular to reject the rendering. Therefore even if an fatal async error happens during the rendering, OOTB we respond with 200 HTTP status from SSR to the client (or save the html file in SSG) - which is bad for SEO, because malformed HTML might be indexed by google

Instead, we should be able to reject from Angular `renderModule` or `renderApplication` and thanks to this return HTTP 500 status from SSR to the client (or don't save the html file in SSG).

Summing up, it's not only timeouts that are important for SEO, but error handling in general too.

FYI in our Angular library Spartacus we're building now an Angular [SSR Error Handling](https://github.com/SAP/spartacus/pull/18742/) feature - a custom wrapper over CommonEngine to be able to populate errors from Angular app to the SSR. Internally we created a custom injection token `PROPAGATE_SERVER_ERROR_RESPONSE = new InjectionToken<Function>` that we call in our custom ErrorHandler class in case of any unhandled exception in the app (+ in case of any HTTP error response from the backend; intercepted in HttpInterceptor).
